android/src/main/java/com/applovin/reactnative/AppLovinMAXAdView.java

```diff
@@ -0,0 +1,85 @@
+package com.applovin.reactnative;
+
+import android.app.Activity;
+import android.content.Context;
+import android.graphics.Color;
+import android.text.TextUtils;
+import android.view.Gravity;
+import android.view.ViewGroup;
+
+import com.applovin.mediation.MaxAdFormat;
+import com.applovin.mediation.ads.MaxAdView;
+import com.applovin.sdk.AppLovinSdkUtils;
+import com.facebook.react.uimanager.ThemedReactContext;
+import com.facebook.react.views.view.ReactViewGroup;
+
+/**
+ * Created by Thomas So on September 27 2020
+ */
+class AppLovinMAXAdView
+        extends ReactViewGroup
+{
+    private final ThemedReactContext reactContext;
+
+    private MaxAdView adView;
+
+    public AppLovinMAXAdView(final Context context)
+    {
+        super( context );
+
+        this.reactContext = (ThemedReactContext) context;
+    }
+
+    @Override
+    public void requestLayout()
+    {
+        super.requestLayout();
+
+        // https://stackoverflow.com/a/39838774/5477988 - This is required to ensure ad refreshes render correctly in RN Android due to known issue
+        if ( adView != null )
+        {
+            adView.measure(
+                    MeasureSpec.makeMeasureSpec( getMeasuredWidth(), MeasureSpec.EXACTLY ),
+                    MeasureSpec.makeMeasureSpec( getMeasuredHeight(), MeasureSpec.EXACTLY )
+            );
+            adView.layout( 0, 0, adView.getMeasuredWidth(), adView.getMeasuredHeight() );
+        }
+    }
+
+    public void maybeAttachAdView(final String adUnitId, final MaxAdFormat adFormat)
+    {
+        Activity currentActivity = reactContext.getCurrentActivity();
+        if ( currentActivity == null )
+        {
+            AppLovinMAXModule.e( "Unable to attach AdView - no current Activity found" );
+            return;
+        }
+
+        // Run after delay to ensure SDK is attached to main module first
```

Comment:
Not really running after a delay, might have to update comment.

Reply:
Ah thx, it used to have a delay which caused a ton of issues